### PR TITLE
Tempus elaboration 29

### PR DESCRIPTION
**Pull request by Himanshu-it**
This PR contains:

1. Creation schema to persist model object definitions and related dao layer services.
2. Creation of CQL upgrade script for model object Definition.
3. Creation of  SQL upgrade script for model object Definition.

### CODE REVIEW

**Comment by jetinder20** [comment]
1. Add validations in service before saving/updating a model like if tenant exits etc (You can refer DeviceServiceImpl, Duplicate name etc for example) 

2. Don’t add the repeating tests in service tests which have already been tested in controllers(Saving model, retrieving) instead you can add corner cases like validations scenario in service as testing them through controller be over kill. 

3. You are not modifying the existing model, so you don’t need an upgrade scripts(add the queries in schema.sq;/cql directly) 

4. Please be consistent with names of services, objects etc. For example BaseModelObjectService could be ModelObjectServiceImpl. There are more places where naming can be better

5. I see potential problems with the model if they reflect what we need on UI.

a. ModelObject needs to have the DataModelId(on branch Tempus_17_and_29).
b. I dont think that we need customer id in that as customerId will have Datamodel already associated to it and model object needs to be associated with Datamodel (Point a covers that)
c.  We dont need "ModelObjectId" in the model explicitly as it can come from the interface.

**Comment by Himanshu-it** [comment]
The recent commit contains, 
1. Adding validations and validation test cases.
2. Renaming modelObject to DataModelObject

File: common/data/src/main/java/com/hashmapinc/server/common/data/DataModelObject/AttributeDefinition.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hashmapinc.server.common.data.DataModelObject;
+
+import com.hashmapinc.server.common.data.id.DataModelObjectId;
+
+import java.util.UUID;
+
+public class AttributeDefinition {
```
Comment (jayeshkapadnis) [inline common/data/src/main/java/com/hashmapinc/server/common/data/DataModelObject/AttributeDefinition.java]: Why isn't this inheriting BaseData ?

**Comment by Himanshu-it** [comment]
In the recent commits:
1. Added a new field "type" in data_model_object table and removed tenant_Id.
2. Removed Id from attribute_definition table, instead used data_model_object_id and name as primary key.
